src/main/java/core/gui/theme/ImageUtilities.java

```diff
@@ -352,369 +352,318 @@ public static ImageIcon getWideImageIcon4Veraenderung(int value, boolean current
-				break;
-			}
-
-			case IMatchRoleID.substCD1: {
-				trickotfarbe = ThemeManager.getColor(HOColorName.SHIRT_SUBDEFENCE);
-				break;
-			}
-
-			case IMatchRoleID.substIM1: {
-				trickotfarbe = ThemeManager.getColor(HOColorName.SHIRT_SUBMIDFIELD);
-				break;
-			}
-
-			case IMatchRoleID.substWI1: {
-				trickotfarbe = ThemeManager.getColor(HOColorName.SHIRT_SUBWING);
-				break;
-			}
-
-			case IMatchRoleID.substFW1: {
-				trickotfarbe = ThemeManager.getColor(HOColorName.SHIRT_SUBFORWARD);
-				break;
-			}
-
-			default:
-				trickotfarbe = ThemeManager.getColor(HOColorName.SHIRT);
-		}
-		return trickotfarbe;
-	}
-
-	public static SVGIcon getSvgIcon(String image) {
-		return (getSvgIcon(image, 24, 24));
-	}
-
-	public static SVGIcon getSvgIcon(String key, int width, int height) {
-		URI svgURI = null;
-		SVGIcon icon = new SVGIcon();
-
-		try {
-			Object imagePath = ThemeManager.getIconPath(key);
-			if (imagePath != null && imagePath instanceof String) {
-				svgURI = ImageUtilities.class.getClassLoader().getResource((String)imagePath).toURI();
-				if (svgURI != null) {
-					icon.setSvgUniverse(new SVGUniverse());
-					icon.setPreferredSize(new Dimension(width, height));
-					icon.setAutosize(SVGIcon.AUTOSIZE_BESTFIT);
-					icon.setInterpolation(INTERP_BILINEAR);
-					icon.setAntiAlias(true);
-					icon.setSvgURI(svgURI);
-					return icon;
-				}
-			}
-		} catch (Exception e) {	}
-
-		// To avoid infinite loop but will never happen
-		if (!key.equals(HOClassicSchema.EMPTY_SVG))
-			return 	getSvgIcon(HOClassicSchema.EMPTY_SVG);
-		else
-			return icon; // generate an exception, will never happen
-	}
-
-	public static double getBrightness(Color colour) {
-		return Math.sqrt(0.241 * colour.getRed()*colour.getRed()
-				+ 0.691 * colour.getGreen()*colour.getGreen()
-				+ 0.068 * colour.getBlue()*colour.getBlue());
-	}
-
-	private static String getHexColor(Color colour) {
-		return "#" + String.format("%1$02X", colour.getRed()) +
-				String.format("%1$02X", colour.getGreen()) +
-				String.format("%1$02X", colour.getBlue());
-	}
-
-	public static Color getColorFromHex(String hexColour) {
-		return new Color(Integer.valueOf(hexColour.substring(1, 3), 16),
-				Integer.valueOf(hexColour.substring(3, 5), 16),
-				Integer.valueOf(hexColour.substring(5, 7), 16));
-	}
+        return getJerseyIcon(posid, taktik, trickotnummer, 20);
+    }
+
+    public static Icon getJerseyIcon(int posid, byte taktik, int trickotnummer, int size) {
+        String key = "trickot_" + posid + "_" + taktik + "_" + trickotnummer + "_" + size;
+        Icon komplettIcon = ThemeManager.getIcon(key);
+
+        if (komplettIcon == null) {
+            Color jerseyColor = getJerseyColorByPosition(posid);
+
+            double brightness = ImageUtilities.getBrightness(jerseyColor);
+            Color textColor = brightness < 130 ? Color.WHITE : Color.BLACK;
+
+            Map<Object, Object> colorMap = Map.of("jerseyColor", jerseyColor,
```

Comment:
So `IconLoader` can handle mapping SVG IDs to their fill colour for us?    That's nice!

Reply:
Also this is essentially a no-op until the icon actually gets used.